### PR TITLE
'unet_simple' -> 'layered_conv' arg in 'unet'

### PR DESCRIPTION
**Pull request by nirolada**

### CODE REVIEW

File: cytounet/model.py [hunk 1]
```diff
@@ -25,63 +24,27 @@ def dice_coef(y_true, y_pred, smooth=1):
...
-    model.compile(optimizer=optimiser_list[optimiser],
-                  loss=loss, metrics=metrics)
-    # model.name = model_name
+    return 1 - dice_coef(y_true, y_pred)
 
-    if pretrained_weights is not None:
-        model.load_weights(pretrained_weights)
 
-    return model
+def apply_conv(input_tile, filters, kernel_regularizer, layered_conv: bool):
```
Comment (Nelson-Gon) [inline cytounet/model.py:34]: Thanks for this, I have tried running but the results are not pretty. Something I did not understand from the code is the `return Conv2D` if double layered. Did you intend to set:

```python
if layered_conv:
    result = Conv2D(filters, 3, activation='relu', kernel_initializer='he_normal', padding='same',
                      kernel_regularizer=kernel_regularizer)(result)
```

Otherwise it seems now for the `double_layered` it returns one non batch normalized `Conv2D` layer.

**Review by Nelson-Gon — COMMENTED** [review]
Hi, please see the comments.

File: cytounet/model.py (hunk 1 above)
Comment (Nelson-Gon) [inline cytounet/model.py:34]: This is the part I intended to quote.

File: cytounet/model.py [hunk 2]
```diff
@@ -25,63 +24,27 @@ def dice_coef(y_true, y_pred, smooth=1):
...
+    result = Conv2D(filters, 3, activation='relu', kernel_initializer='he_normal', padding='same',
+                    kernel_regularizer=kernel_regularizer)(input_tile)
+    result = BatchNormalization()(result)
+    if layered_conv:  # applies double-layered conv
+        return Conv2D(filters, 3, activation='relu', kernel_initializer='he_normal', padding='same',
+                      kernel_regularizer=kernel_regularizer)(result)
```
Comment (Nelson-Gon) [inline cytounet/model.py:40]: Meant to quote these lines, sorry.

Comment (nirolada) [reply cytounet/model.py:40]: All this function does:
1. Apply a convolution
2. Apply batch normalization
3. If 'layered_conv' is True, apply a convolution once again and return it immediately
4. If 'layered_conv' is False, return the result after the batch normalization

File: cytounet/model.py (hunk 1 above)
Comment (nirolada) [reply cytounet/model.py:34]: Read the following comment after you read my comment to lines 38-40:
Maybe I shouldn't have applied batch_normalization if 'layered_conv' is False (simple model)?

File: cytounet/model.py (hunk 2 above)
Comment (Nelson-Gon) [reply cytounet/model.py:40]: Ah, I see. Was confused by the double return but a second look makes it clearer. Thanks!

Comment (nirolada) [reply cytounet/model.py:40]: Make sure to see my comment to your other review please.

File: cytounet/model.py (hunk 1 above)
Comment (Nelson-Gon) [reply cytounet/model.py:34]: I think it's fine to apply a `BatchNorm` there.

File: cytounet/model.py (hunk 2 above)
Comment (Nelson-Gon) [reply cytounet/model.py:40]: Sorry, I had missed it.
